overlays/firefox-darwin/default.nix

```diff
@@ -0,0 +1,6 @@
+{ channels, firefox-darwin, ... }:
+
+final: prev: {
+  # Access with `pkgs.firefox-darwin.<name>`.
+  firefox-darwin = firefox-darwin.overlay final prev;
```

Comment:
This overlay is a bit unique since it seems to override the existing package entirely, replacing it with one that _only_ supports darwin. In order to provide both in the same package set, I've placed the overlay result on a new attribute.